### PR TITLE
feat: Display better error message for malformed anchor requests

### PR DESCRIPTION
**Pull request by ukstv**

### CODE REVIEW

**Review by v-stickykeys — APPROVED** [review]
Overall I understand what is happening and the changes made though I'm less confident about the specifics of the decoding logic (e.g. root being left means failure)

File: src/ancillary/anchor-request-params-parser.ts [hunk 1]
```diff
@@ -22,19 +26,74 @@ export const RequestAnchorParamsV1 = t.intersection(
...
+        cid: root.tip,
+        genesisFields: genesisFields,
+      })
+    } catch (e: any) {
+      const message = e.message || String(e)
+      return t.failure(bytes, context, `Can not decode CAR file: ${message}`)
+    }
+  }
+
+  private retrieveGenesisRecord(genesisCid: CID, carFile: CAR): unknown {
```
[the remark this answers is not in the document]
Comment (ukstv) [reply src/ancillary/anchor-request-params-parser.ts:84]: Here we return an IPLD record from CAR file, which is not quite trusted. We do not know yet what's inside. The record might contain garbage. That's why unknown.

Validation is on https://github.com/ceramicnetwork/ceramic-anchor-service/pull/983/files/20021855a24422fdffb9a09e38ac9d3f10de23e7#diff-b1dbe36cf289ff428fcee2fdbbd124a2177a302d17fbe0ab760c960df98a952bR69

Only after it is decoded, we can be sure, it is a genesis record.

**Comment by ukstv** [comment]
"root being left means failure" => It is part of `Either` monad used by io-ts. `Either<TError, TValue> = Left<TError> | Right<TValue>`. Throwing/catching an error is quite expensive. Instead of throwing an error you return `Left<Error>`. Then you check if `isLeft(returntedValue)` to find if it is an error. `isRight(returnedValue)` on the opposite indicates that `returnedValue` contains proper value without any errors.

**Comment by ukstv** [comment]
@v-stickykeys https://blog.logrocket.com/javascript-either-monad-error-handling/

**Pull request merged**